### PR TITLE
AppTour and Rating components - AB#15749

### PR DESCRIPTION
**Pull request by BronzBierd**
# Implements [AB#15749](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15749)

## Description
1. Convert AppTourComponent
2. Convert RatingComponent

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [X] Not Required

### AppTour Desktop:
image

### AppTour Mobile:
height must be a fixed height for vuetify carousel to work without breaking transitions.

The slight "bleed" around the image change from the top to the sides pending on the width of the screens so there isn't a perfect solution for mobile, this seems to be a frustrating aspect of v-img. Can revisit this with native  later once Mandad has explored the implementation.
image

### Rating Desktop:
image

### Rating Mobile:
image

## Notes
v-dialog, doesn't natively provide a card like structure so one must define your own structure, a straight row with a card directly within worked well, there may be a better layout option but did not invest more time into it for a marginal improvement.

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/WebClient/src/NewClientApp/src/components/modal/AppTourComponent.vue [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    slideIndex.value = 0;
+    isVisible.value = false;
+}
+
+function next(): void {
+    slideIndex.value++;
+}
+function previous(): void {
+    slideIndex.value--;
+}
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/components/modal/AppTourComponent.vue:103]:
```suggestion
function next(): void {
    slideIndex.value++;
}

function previous(): void {
    slideIndex.value--;
}
```

File: Apps/WebClient/src/NewClientApp/src/components/modal/AppTourComponent.vue [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+                            :src="slide.imageUri"
+                            :alt="slide.imageAlt ?? slide.title"
+                            transition="scroll-x-reverse-transition"
+                        />
+                    </v-carousel>
+                </v-card-text>
+                <template #actions>
+                    <v-container>
+                        <v-row v-if="currentSlide">
+                            <v-col>
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/components/modal/AppTourComponent.vue:149]: Row & col wrapper here don't seem to be needed - could be replaced with `<div>`.

File: Apps/WebClient/src/NewClientApp/src/components/modal/AppTourComponent.vue [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+                                    variant="primary"
+                                    class="ml-4"
+                                    data-testid="app-tour-next"
+                                    @click="next()"
+                                    text="Next"
+                                />
+                            </v-col>
+                        </v-row>
+                        <v-row v-else class="mt-6">
+                            <v-col class="d-flex justify-center">
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/components/modal/AppTourComponent.vue:208]: Row & col wrapper here don't seem to be needed - could be replaced with `<div>`.

File: Apps/WebClient/src/NewClientApp/src/components/modal/RatingComponent.vue [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            hideModal();
+            emit("on-close");
+        });
+}
+</script>
+
+<template>
+    <v-dialog
+        id="rating-modal"
+        ref="rating-modal"
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/components/modal/RatingComponent.vue:80]:
```suggestion
```
This ref doesn't seem to be used.

**Review by MikeLyttle — APPROVED** [review]

**Review by BrianMaki — APPROVED** [review]

**Pull request merged**